app-crypt/certbot/certbot-3.2.0-r102.ebuild

```diff
@@ -0,0 +1,256 @@
+	certbot-apache? (
+		dev-python/python-augeas[${PYTHON_USEDEP}]
+	)
+	certbot-dns-dnsimple? (
+		>=dev-python/dns-lexicon-3.14.1[${PYTHON_USEDEP}]
+	)
+	certbot-dns-dnsmadeeasy? (
+		>=dev-python/dns-lexicon-3.14.1[${PYTHON_USEDEP}]
+	)
+	certbot-dns-gehirn? (
+		>=dev-python/dns-lexicon-3.14.1[${PYTHON_USEDEP}]
+	)
+	certbot-dns-google? (
+		>=dev-python/google-api-python-client-1.6.5[${PYTHON_USEDEP}]
+		>=dev-python/google-auth-2.16.0[${PYTHON_USEDEP}]
+	)
+	certbot-dns-linode? (
+		>=dev-python/dns-lexicon-3.14.1[${PYTHON_USEDEP}]
+	)
+	certbot-dns-luadns? (
+		>=dev-python/dns-lexicon-3.14.1[${PYTHON_USEDEP}]
+	)
+	certbot-dns-nsone? (
+		>=dev-python/dns-lexicon-3.14.1[${PYTHON_USEDEP}]
+	)
+	certbot-dns-ovh? (
+		>=dev-python/dns-lexicon-3.15.1[${PYTHON_USEDEP}]
+	)
+	certbot-dns-rfc2136? (
+		>=dev-python/dnspython-2.6.1[${PYTHON_USEDEP}]
+	)
+	certbot-dns-route53? (
+		>=dev-python/boto3-1.15.15[${PYTHON_USEDEP}]
+	)
+	certbot-dns-sakuracloud? (
+		>=dev-python/dns-lexicon-3.14.1[${PYTHON_USEDEP}]
+	)
+	certbot-nginx? (
+		>=dev-python/pyopenssl-25.0.0[${PYTHON_USEDEP}]
+		>=dev-python/pyparsing-2.4.7[${PYTHON_USEDEP}]
+	)
+	selinux? ( sec-policy/selinux-certbot )
+"
+# RDEPEND+="
+#	!<app-crypt/certbot-dns-cloudflare-3.2.0-r100
+#
+# 	>=dev-python/requests-toolbelt-0.3.0[${PYTHON_USEDEP}] # @TODO is still necessary?
+# 	certbot-dns-cloudflare? (
+# 		# Available in GURU
+# 		>=dev-python/cloudflare-2.19[${PYTHON_USEDEP}]
+# 		<dev-python/cloudflare-2.20[${PYTHON_USEDEP}]
+# 	)
+# 	certbot-dns-digitalocean? (
+# 		# Available in GURU
+# 		>=dev-python/digitalocean-1.11[${PYTHON_USEDEP}]
+# 	)
+# "
+
+# Note: "docs" is not an actual directory under "S", they are actually
+# under each modules, see python_compile_all redefinition, but keep
+# this instruction enabled for dependency configuration.
+distutils_enable_sphinx docs \
+	dev-python/sphinx-rtd-theme
+distutils_enable_tests pytest
+
+# Used to build documentation
+python_compile_all() {
+	use doc || return
+
+	sphinx_compile_all
+
+	# Subdirectory "_build/html" from build_sphinx in eclass/python-utils-r1.eclass
+	mv "${_DISTUTILS_SPHINX_SUBDIR}/_build/html" "${temp_docs}/${dir}" || die
+}
+
+distutils-r1_src_compile() {
+	debug-print-function ${FUNCNAME} "$@"
+	local ret=0
+
+	# Used to build documentation
+	use doc && {
+		# Stores temporary modules docs in each subdirectories, will be used for HTML_DOCS
+		local temp_docs="${T}/docs"
+		mkdir "${temp_docs}" || die
+	}
+
+	local certbot_dirs=()
+	local base module dir
+	for base in "${CERTBOT_BASE[@]}"; do
+		certbot_dirs+=("${base}")
+	done
+	for module in "${CERTBOT_MODULES_EXTRA[@]}"; do
+		use "certbot-${module}" \
+			&& certbot_dirs+=("certbot-${module}")
+	done
+
+	for dir in "${certbot_dirs[@]}"; do
+		pushd "${dir}" > /dev/null || die
+		local BUILD_DIR=${dir%%/}
+
+		_distutils-r1_run_foreach_impl distutils-r1_python_compile || ret=${?}
+		_distutils-r1_run_common_phase python_compile_all || ret=${?}
```

Comment:
Don't reinvent the eclass. Not calling the phases multiple times also means "don't inline the stuff you aren't supposed to call multiple times and call it".

Reply:
I forgot to mention, I did this because I found no solution to call `_distutils-r1_post_python_compile` (again, an internal function) with a way to have a correct `root=${BUILD_DIR}/install`, and also called multiple times for each "module".

Otherwise the install phase fails during `distutils-r1_python_install`, as “we are assuming that _distutils-r1_post_python_compile has been called [and ${root} has not been altered since]”.